openstates/me/people.py

```diff
@@ -31,75 +31,99 @@ def scrape(self, chamber=None):
             yield from self.scrape_reps()
 
     def scrape_reps(self):
-        url = 'http://www.maine.gov/legis/house/dist_mem.htm'
-        page = self.get(url).text
-        page = lxml.html.fromstring(page)
-        page.make_links_absolute(url)
-
-        # These do not include the non-voting tribal representatives
-        # They do not have numbered districts, and lack a good deal of
-        # the standard profile information about representatives
-        for district in page.xpath('//a[contains(@href, "dist_twn")]/..'):
-            if "- Vacant" in district.text_content():
-                self.warning("District is vacant: '{}'".
-                             format(district.text_content()))
-                continue
-
-            _, district_number = district.xpath('a[1]/@href')[0].split('#')
-
-            leg_url = district.xpath('a[2]/@href')[0]
-            leg_info = district.xpath('a[2]/text()')[0]
-
-            INFO_RE = r'''
-                    Representative\s
-                    (?P<member_name>.+?)
-                    \s\(
-                    (?P<party>[DRCUIG])
-                    -
-                    (?P<district_name>.+?)
-                    \)
-                    '''
-            info_search = re.search(INFO_RE, leg_info, re.VERBOSE)
-
-            if not info_search:
-                leg_url = district.xpath('a[3]/@href')[0]
-                leg_info_second_active = district.xpath('a[3]/text()')[0]
-
-                member_name = leg_info_second_active
-                mem_info = district.xpath('a[3]/following-sibling::text()')
-                party = _party_map[mem_info[0][2]]
-                district = mem_info[0].split('-')[1][:-1]
-            else:
-                member_name = info_search.group('member_name')
-                party = _party_map[info_search.group('party')]
-                district_name = info_search.group('district_name')
+        mapping = {
+            'district': 0,
+            'last_name': 1,
+            'first_name': 2,
+            'mid_name': 3,
+            'jr_sr': 4,
+            'party': 5,
+            'legal_res': 6,
+            'address1': 7,
+            'address2': 8,
+            'town': 9,
+            'state': 10,
+            'zip_code': 11,
+            'zip_plus4': 12,
+            'home_phone': 13,
+            'cell_phone': 14,
+            'business_phone': 15,
+            'other_phone': 16,
+            'fax': 17,
+            'email': 18,
+            'seat_no': 19,
+            'house_leadership': 20,
+            'committees': 21
+        }
 
-            # Get the photo url.
-            html = self.get(leg_url).text
-            doc = lxml.html.fromstring(html)
-            doc.make_links_absolute(leg_url)
-            (photo_url,) = doc.xpath('//img[contains(@src, ".jpg")]/@src')
+        url = (
```

Comment:
This can be on its own line, outside of parentheses. Also, make sure to `Person.add_source()` for this URL, so that it's noted as a primary data source in the database.

Reply:
Also, it looks like this URL would always return 2017–2018 legislators. Can you replace `128` in the URL with `self.latest_session`?

Reply:
Not your purview, but can you also replace `128` with `self.latest_session` for the upper-house scraping URL, later in this Python module?

Reply:
The url for the spreadsheet for the upper-house has been placed inside parentheses. Is there a reason behind that? Is it because the line is crossing a certain number of characters and needs to be split?